src/com/softwaret/kpdf/interactor/register/RegisterInteractor.kt

```diff
@@ -1,8 +1,14 @@
 package com.softwaret.kpdf.interactor.register
 
 import com.softwaret.kpdf.model.UserTile
+import com.softwaret.kpdf.repository.validation.input.login.LoginValidator
+import com.softwaret.kpdf.repository.validation.input.name.NameValidator
+import com.softwaret.kpdf.repository.validation.input.password.PasswordValidator
 
-interface RegisterInteractor {
+interface RegisterInteractor :
+    LoginValidator,
```

Comment:
To mi się średnio podoba, bo
- walidujemy zazwyczaj pod to samo (blank, null itp.)
- każdy interactor musi wtedy implementować te interfejsy (co może doprowadzić do powtórzeń, bo `LoginInteractor` i `RegisterInteractor` beda mialy te same metody - które są powtarzalne)
- walidacja raczej nie powinna być odpowiedzialnością `interactora` raczej on powinien juz operowac na nadających się do czegoś danych
Myślę czy nie zrobić czegoś takiego jak `InputValidator`, wstrzykiwany do kontrolera i wtedy walidować odpowiednie pola, a poźniej wykonywać akcje na interactorze gdy zwalidowane pola są ok.
Plusy rozwiazania są takie 
- mniejszy boilprate w budowie interactorów
- mniejszy boilprate, bo można cale sprawdzanie czy jest `blank` zamknąć w jednym walidatorze, bo tak naprawdę nie obchodzi nas jakie pole jest błędne (bo gdy pole jest `blank` przy logowaniu dajemy po prostu 401, tak jak rozmawialiśmy, bez informacji o tym czy login czy haslo bylo `blank`)
- Jeden prosty `PRZETESTOWANY` serwis wstrzykiwany z kodeiny, zamiast powtarzalnej delegacji przy kazdym use case
Serwis można nawet zamknąć za interfejsem i wstrzykiwać mu mniejsze walidatory (do nulla, do blank, czy np do zawierania w sobie małych i duzych liter w przypadku hasła) w zależności od przypadku użycia